src/sparkx/Lattice3D.py

```diff
@@ -1026,19 +1032,33 @@ def add_particle_data(self, particle_data, sigma, quantity, add = False):
                 k_min = 0
                 k_max = self.num_points_z_
 
+            norm=0
+
             for i in range(i_min, i_max):
                 for j in range(j_min, j_max):
                     for k in range(k_min, k_max):
                         # Get the coordinates of the current grid point
                         xi, yj, zk = self.get_coordinates(i, j, k)
 
                         # Calculate the value to add to the grid at (i, j, k)
-                        smearing_factor = kernel.pdf([xi, yj, zk])
+                        if(kernel == "gaussian"):
+                            smearing_factor = kernel_value.pdf([xi, yj, zk])
+                        else:
```

Comment:
Is the following part similar to the procedure in vhlle?

Reply:
What exactly do you mean?

Reply:
I mean the implementation of the covariant smearing with the formula from [https://indico.cern.ch/event/1043736/contributions/5441985/attachments/2672029/4632166/IS_flashtalk_zpaulinyova.pdf](url) on slide 4, as it is implemented in vhlle.

Reply:
Yes, that is the same formula.

Reply:
Ok, then I have to check it again. It does not look like it at the first look.

Reply:
The only difference is that we are working here in Minkowski coordinates and not in Milner.

Reply:
It might be that we need a conversion of the Minkowski coordinates to Milne coordinates in the end, depending on the hydro code.

Reply:
We decided to put the transformation in a different PR